_posts/2023-03-16-back-to-terraform.md

```diff
@@ -0,0 +1,226 @@
+
+Adding to the complexity, the Terragrunt configuration was loading different versions of the Terraform modules in different environments, and the Terraform state was split into modules as well, with custom scripts to read and manipulate the state. All of these factors made it difficult for the new team to make progress on the project, and they began to question the decision to use Terragrunt in the first place.
+
+### Deciding to Return to Terraform
+
+After struggling to work with the complex and fragmented Terragrunt setup, the new infrastructure team wanted to find a simpler, more streamlined approach. They sought a setup with a low threshold, low maintenance, and easy to understand.
+
+As they dug deeper into the Terragrunt configuration, they found that even the company's architects had some doubts about its usefulness. So they began discussing alternatives, and eventually someone asked the question: "Why are we using Terragrunt?"
+
+The answer was surprising: the team wasn't even using any of Terragrunt's real advantages, and was instead dealing with unnecessary overhead. The configuration of all external modules was just a single key-value file, with no clear indication of which value was passed to which module.
+
+It became clear that continuing to use Terragrunt was simply pointless, and only adding to the team's frustration. They, together with the company's architects, made the decision to switch back to using Terraform directly, in order to simplify their workflow and make progress on the project.
+
+### Returning to terraform without breaking the existing infrastructure
+
+After making the decision to switch back to using Terraform directly, the team realized that a complete overhaul of their infrastructure setup would be necessary. Rather than throwing out all of the existing Terragrunt code, however, they decided to take a hybrid approach.
+
+The first step was to set up a new mono git repository that would contain all of the different Terraform modules that had previously been spread across multiple repositories. They also created a new Terraform root configuration setup, which would enable them to manage the entire infrastructure as a single entity.
+
+Next, the team began moving these Terraform modules into their new monorepo. As they did so, they took the opportunity to clean them up, removing any unused input variables or features. They also improved the security setup, as the project required the infrastructure to be publicly accessible instead of hidden behind an on-premises network.
```

Comment:
```suggestion
Next, the team began moving these Terraform modules into their new monorepo. 
As they did so, they took the opportunity to clean them up, removing any unused input variables or features. 
They also improved the security setup, as the project required the infrastructure to be publicly accessible instead of hidden behind an on-premises network.
```